sunpy/util/util.py

```diff
@@ -195,3 +197,32 @@ def hash_file(path):
             sha256.update(data)
 
     return sha256.hexdigest()
+
+
+def get_rectangle_coordinates(bottom_left, *, top_right = None, width: u.deg = None, height: u.deg = None):
+
+    if (not isinstance(bottom_left, (SkyCoord, BaseCoordinateFrame)) and 
+        not isinstance(top_right, (None, SkyCoord, BaseCoordinateFrame))):
+        raise TypeError("bottom_left and top_right must be of type SkyCoord or BaseCoordinateFrame")
+
+    elif bottom_left.shape == (2,):
+        top_right = bottom_left[1]
+        bottom_left = bottom_left[0]
+
+    elif top_right is not None:
+        width = top_right.spherical.lon - bottom_left.spherical.lon   # Getting the difference in Longitudes.
+        height = top_right.spherical.lat - bottom_left.spherical.lat  # Getting the difference in Latitudes.
+
+        if width <= 0*u.deg or height <= 0*u.deg:
+            raise ValueError("top_right must be to the right and above bottom_left.")
+
+    elif width is not None and height is not None:
+        top_right = SkyCoord(bottom_left.spherical.lon + width,
+                             bottom_left.spherical.lat + height,
+                             frame=bottom_left.frame)
+
+    else:
+        raise ValueError("Invalid input, either bottom_left and top_right \
+                         or bottom_left and height and width must be provided")
+
+    return (bottom_left, top_right)
```

Comment:
multiple return is implicitly a tuple
```suggestion
    return bottom_left, top_right
```